### PR TITLE
Convert device-api/v1 from coffeescript to javascript

### PR DESCRIPTION
**Pull request by Page-**

### CODE REVIEW

File: src/lib/validation.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export function checkString(s: unknown): string | void {
  * Given a value which can be a string, boolean or number, return a boolean
  * which represents if the input was truthy
  */
-export function checkTruthy(v: unknown): boolean | void {
+export function checkTruthy(v: unknown): boolean | undefined {
```
Comment (Page-) [inline src/lib/validation.ts]: Specifying this as `undefined` means we can use `??` like `checkTruthy(x) ?? false`

File: src/device-api/v1.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+import * as constants from '../lib/constants';
+import { checkInt, checkTruthy } from '../lib/validation';
+import { doRestart, doPurge, serviceAction } from './common';
+
+export const createV1Api = function(router, applications) {
+	const { eventTracker } = applications;
+
+	router.post('/v1/restart', function(req, res, next) {
+		const appId = checkInt(req.body.appId);
+		const force = checkTruthy(req.body.force) ?? false;
```
Comment (Page-) [inline src/device-api/v1.js]: I had to add the `?? false` here to pass the type checks, previously `force` could be `undefined` - is `false` the right default/

File: src/device-api/v1.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+					appToSend.commit = status.commit;
+				}
+				return res.json(appToSend);
+			},
+		).catch(next);
+	});
+
+	router.post('/v1/purge', function(req, res, next) {
+		const appId = checkInt(req.body.appId);
+		const force = checkTruthy(req.body.force) ?? false;
```
Comment (Page-) [inline src/device-api/v1.js]: I had to add the `?? false` here to pass the type checks, previously `force` could be `undefined` - is `false` the right default/

File: src/device-api/v1.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+			return res.status(400).send('Missing app id');
+		}
+		return doRestart(applications, appId, force)
+			.then(() => res.status(200).send('OK'))
+			.catch(next);
+	});
+
+	const v1StopOrStart = function(req, res, next, action) {
+		const appId = checkInt(req.params.appId);
+		const force = checkTruthy(req.body.force) ?? false;
```
Comment (Page-) [inline src/device-api/v1.js]: This `?? false` wasn't technically required by typescript but it seemed the right thing to do based on the two cases it did catch

File: src/device-api/v1.js (hunk 2 above)
Comment (CameronDiver) [reply src/device-api/v1.js]: It is yeah.

File: src/device-api/v1.js (hunk 4 above)
Comment (CameronDiver) [reply src/device-api/v1.js]: Yeah, the `checkTruthy` function is a mess, but I'd like the entire codebase to be in typescript before changing fundamentals like that.

**Pull request merged**